### PR TITLE
Remove MPI wrappers for FMS and MAPL builds to enable apple-clang%15.x builds

### LINKED ISSUE #1083: FMS and MAPL builds fail on MacOS Sonoma 14 when using Xcode 15.1
**Describe the bug**

FMS and MAPL builds fail due to the inability to find the OpenMP Fortran component. Both of these packages use the mpi wrappers and when I removed the usage of the mpi wrappers, the builds completed successfully. I did try a variety of other solutions including exporting OpenMP_ROOT which did not fix the problem. Both of these packages use CMake which should be fine building without the use of the wrappers, so there is a general desire to not use the mpi wrappers.

**To Reproduce**

On a Mac with Sonoma installed, and with Xcode 15.1 (apple-clang@15.0.0) and GNU Fortran 12.3.0, follow the instructions in the spack-stack doc for building on the MacOS platform.

**Expected behavior**

The build described above should complete successfully.

**System:**

See above

**Additional context**

### PR DESCRIPTION
**Pull request by srherbener**
### Summary

This PR is simply for syncing up the spack submodule hash for the spack PR (JCSDA/spack/pull/431) that enables builds of FMS and MAPL using apple-clang%15.x. 

### Testing

These changes were tested on my Mac using apple-clang%15.0.0 (Xcode 15.1). 

### Applications affected

FMS and MAPL are impacted. The usage of the MPI wrappers has been removed.

### Systems affected

The MPI wrappers were removed unconditionally so these changes can affect the HPC platforms, plus MacOS, using FMS and MAPL.

### Dependencies

- [x] waiting on JCSDA/spack/pull/431

### Issue(s) addressed

Fixes #1083 

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications. (Mac, several NOAA systems, see https://github.com/spack/spack/pull/43726 for details)
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by climbfuji** [comment]
Looks good. Will approve after the spack PR is merged etc.

**Comment by srherbener** [comment]
Ready for review. spack PR has been merged, and the spack submodule hash is updated here (and the .gitmodules file has been restored).

**Review by climbfuji — APPROVED** [review]

**Pull request merged**
